### PR TITLE
Keep retrying mysql port if it is 0 in Vttablets

### LINKED ISSUE #12985: Bug Report: VTOrc doesn't fix replication on a tablet that was just restored
### Overview of the Issue

It was noticed that VTOrc was unable to fix replication on a tablet which was recently restored. 

The issue actually isn't in VTOrc at all!
We see in the logs - 
```
Discovered: alias:{cell:"a" uid:1} hostname:"10.10.10.10" port_map:{key:"grpc" value:15999} port_map:{key:"vt" value:15000} keyspace:"db" shard:"-" key_range:{} type:REPLICA db_name_override:"db" mysql_hostname:"10.10.10.10" db_server_version:"8.0.30" default_conn_collation:254
```
Notice that `mysql_port` isn't in the log.

Because `mysql_port` value is set to 0, VTOrc is unable to run any repairs since it thinks that this configuration of mysql host and port isn't a valid configuration.

On the vttablet, we see logs like - 
```
Cannot get current mysql port, will keep retrying every 1s: net.Dial(/vt/socket/mysql.sock) to local server failed: dial unix /vt/socket/mysql.sock: connect: no such file or directory (errno 2002) (sqlstate HY000)
```
and eventually - 
```
Identified mysql port: 0
```

This is incorrect since we end up displaying the port to 0.

### Reproduction Steps

1. Have a running cluster with VTOrc
2. Bring up a tablet that is restored from backup
3. Wait and see if the replication is fixed or not.
4. Repeat until failure is noticed

### Binary Version

```sh
main
```

### Operating System and Environment details

```sh
main
```

### Log Fragments

_No response_

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR fixes the issue described in #12985 by keeping on retrying until we see a non-0 port in vttablet.

## Related Issue(s)

- Fixes #12985 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by dbussink — APPROVED** [review]
I guess due to the racy nature here it's very hard to add a specific test for this?

**Comment by GuptaManan100** [comment]
@dbussink I was able to add a test. There was actually a test that already existed for the function. I augmented it to test the new behaviour.

File: go/vt/vttablet/tabletmanager/tm_init_test.go [hunk 1]
```diff
@@ -422,9 +422,10 @@ func TestStartCheckMysql(t *testing.T) {
 	assert.Equal(t, "foo", ti.MysqlHostname)
 }
 
+// TestStartFindMysqlPort tests the functionality of findMySQLPort on tablet startup
```
Comment (frouioui) [inline go/vt/vttablet/tabletmanager/tm_init_test.go:425]: have you made sure this test passes continuously locally after running it a lot of times?

**Review by frouioui — APPROVED** [review]

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vttablet/tabletmanager/tm_init_test.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vttablet/tabletmanager/tm_init_test.go:425]: Yes, it does 👍

**Pull request merged**